src/ui/components/button/PlayButton.tsx

```diff
@@ -46,7 +46,7 @@ export class PlayButton extends PureComponent<PlayButtonProps, PlayButtonState>
     context.player.addEventListener(PlayerEventType.SEEKING, this.onSeeking);
     this.setState({
       paused: context.player.paused,
-      ended: context.player.currentTime === context.player.duration,
+      ended: (context.player.duration > 0) && (context.player.currentTime === context.player.duration),
```

Comment:
Why does `context.player.ended` not work?

Reply:
Would make sense indeed. But I just noticed we don't have this on our API. We're only supporting the ENDED event but never store it as a state.

Reply:
maybe it would make sense to rely on something that's ended (or the property or the event), because I think on iOS we had an issue where the duration wasn't equal to the last currentTime.

Afaik it was fixed and it dispatches the `ended` event (and reflects it on the property too)
Otherwise we reintroduce the same issue on the RN level.

(on the other hand it would be nice if iOS could update the duration properly too, if possible to detect the duration change (if any))

Reply:
maybe it would make sense to rely on something that's ended (or the property or the event), because I think on iOS we had an issue where the duration wasn't equal to the last currentTime.

Afaik it was fixed and it dispatches the ended event (and reflects it on the property too)
Otherwise we reintroduce the same issue on the RN level.

(on the other hand it would be nice if iOS could update the duration properly too, if possible to detect the duration change (if any))